WordPress/wordpress-playground
# playground.wordpress.net: Support mounting local files

Since https://github.com/WordPress/wordpress-playground/pull/547 adds support for OPFS API, I just realized the same API can be used to read/write local files. This PR adds a local directory picker and a "refresh" button to bring the changes back into Playground's MEMFS.

<img width="1232" alt="CleanShot 2023-06-15 at 19 37 10@2x" src="https://github.com/WordPress/wordpress-playground/assets/205419/aa8d6b87-1b7d-4789-a497-201ecadf121a">

## Testing instructions

* Select an empty local directory
* Confirm it populated that with files
* Make a change to the homepage
* Refresh the page
* Confirm you got asked about whether to reuse that directory
* Say yes
* Confirm your homepage change is now visible

## Other considerations

* You can only save and load a Playground instance. The "context detection" logic from `wp-now` could be really handy here. It would have to be refactored to use shared, isomorphic feature that works through `php` methods like `php.readFileAsBuffer()` which means they'll need some enhancements like support for reading the first 8kb.
* A proper OPFS filesystem for Emscripten would remove the need for a `refresh files` button as everything would be stored and read directly from OPFS. #544 started exploring that but it uses the deprecated sync API. We'd need to find a way to use the new async one.
    * On the flip side, a proper OPFS is also quite slow. Maybe the sync button is for the best?
